src/index.js

```diff
@@ -61,126 +66,197 @@ const monitorChromecastPlayer = function (player, options) {
-        player.mux.emit('ratechange');
-        break;
-      case cast.framework.events.EventType.TIME_UPDATE:
-        currentTime = event.currentMediaTime;
-        player.mux.emit('timeupdate');
-        break;
-      case cast.framework.events.EventType.SEGMENT_DOWNLOADED:
-        let now = Date.now();
-        let loadData = {
-          request_event_type: 'SEGMENT_DOWNLOADED',
-          request_start: now - event.downloadTime - 1,
-          request_response_start: now - event.downloadTime,
-          request_response_end: now,
-          request_bytes_loaded: event.size,
-          request_type: 'media'
-        };
-
-        player.mux.emit('requestcompleted', loadData);
-        break;
+          break;
+        case cast.framework.events.EventType.MEDIA_FINISHED:
+        case cast.framework.events.EventType.LIVE_ENDED:
+          if (!videoChanged) {
+            player.mux.emit('ended');
+          }
+          videoChanged = false;
+          break;
+        case cast.framework.events.EventType.MEDIA_STATUS:
+          if (event.mediaStatus.videoInfo !== undefined) {
+            // Note: it appears the videoInfo field is always undefined
+            videoSourceWidth = event.mediaStatus.videoInfo.width;
+            videoSourceHeight = event.mediaStatus.videoInfo.height;
+          }
+          if (event.mediaStatus.media !== undefined &&
+            event.mediaStatus.media.duration !== undefined) {
+            duration = mux.utils.secondsToMs(event.mediaStatus.media.duration);
+          }
+          break;
+        case cast.framework.events.EventType.SEEKING:
+          isSeeking = true;
+          player.mux.emit('seeking');
+          break;
+        case cast.framework.events.EventType.SEEKED:
+          player.mux.emit('seeked');
+          break;
+        case cast.framework.events.EventType.PAUSE:
+          isPaused = true;
+          if (!isSeeking) {
+            // Note:
+            // When a seek happens, cast sdk sends the `cast.framework.events.EventType.PAUSE`.
+            // We need to suppress the player.mux.emit('pause') call, because if it goes out, in our data view, seek
+            // events will become pause event
+            player.mux.emit('pause');
+          }
+          break;
+        case cast.framework.events.EventType.PLAYING:
+          if (isPaused) {
+            player.mux.emit('play');
+          }
+          if (isSeeking) {
+            isSeeking = false;
+          }
+          isPaused = false;
+          player.mux.emit('playing');
+          break;
+        case cast.framework.events.EventType.ERROR:
+          if (!options.automaticErrorTracking) { return; }
+          switch (event.detailedErrorCode) {
+            case 901:
+            case 902:
+              // specific errors to ads
+              player.mux.emit('aderror');
+              break;
+            case 903:
+            case 904:
+            // do nothing since these aren't fatal errors
```

Comment:
Love catching this... just indent the comment (yeah yeah I know I'm OCD sometimes)